### DISCUSSION TITLE
Switching from Streaming to Pre-Recorded API

### DISCUSSION THREAD

**Discussion by Naillik1** [deepgram/community#318] [General help] [answered]
Hi, I am trying to switch my application from the [streaming API](https://developers.deepgram.com/reference/streaming) to the [pre-recorded API](https://developers.deepgram.com/reference/pre-recorded).

I would like to know how to construct a request to the pre-recorded API that's as similar as possible to my current use of the streaming API.

Currently, my application uses the Deepgram streaming API, and I've been quite happy with the transcription quality. 
I initiate a websocket connection with the URL `wss://api.deepgram.com/v1/listen?channels=1&encoding=linear16&sample_rate=16000&tier=nova` , and send frames of ~4096 bytes containing ~150ms of audio.

I've attempted to convert this to a regular HTTP request per the docs:
```elixir
%{
  scheme: :https,
  host: "api.deepgram.com",
  method: "POST",
  path: "/v1/listen",
  headers: [
    {"Authorization", "Token <API_KEY>"},
    {"Content-Type", "audio/wav"}
  ],
  body: <<raw binary of ~4096 bytes (~150ms of audio)>>,
  query: "channels=1&encoding=linear16&sample_rate=16000&tier=nova"
}
```

However, this always yields an error response despite no changes to the format of the audio being delivered:
```json
{
  "err_code": "Bad Request",
  "err_msg": "Bad Request: failed to process audio: corrupt or unsupported data",
  "request_id": "<request_id>"
}
```

I'm wondering how I can switch from the streaming API to the pre-recorded API?
Any help is greatly appreciated!

**Comment by jjmaldonis** [comment, marked as answer]
Hey @Naillik1 it looks like you posted in our Discord as well and you and Sandra got this working.

For anyone else reading this, the solution was to attach a WAV header to the audio chunks when sending the broken-up audio files to Deepgram's prerecorded endpoint.

**Reply by Naillik1** [reply]
Yep, consider this answered!
~I needed to prepend a proper header for my audio format before sending my data in smaller chunks.~ 
Edit: answer included above
